### ISSUE TITLE
[Feature Request]  Link aspects, stunts and extras to chat

### ISSUE THREAD

**Issue by ToffeeFi** [anvil-vtt/FateX#133]
Maybe I'm a little inattentive, but I've never seen anyone ask for this feature:
I'd like to be able to link aspects with their descriptions, stunts, extras to chat T.T

Can't find any module to do this, and I am not a developer to code this function myself.

**Comment by Alexandre-nk-Perdereau** [comment]
I'm currently working on it. For now, it looks like this:

<img width="1344" height="1086" alt="Image" src="https://github.com/user-attachments/assets/55c186e9-bbc9-4eb2-adf3-285b83d42d66" />

<img width="1347" height="1078" alt="Image" src="https://github.com/user-attachments/assets/14f3cb88-a486-42a6-919f-04e72b9c21fe" />

<img width="437" height="172" alt="Image" src="https://github.com/user-attachments/assets/1cd9ec81-c78e-44de-baf8-9aa73964a91d" />

<img width="434" height="191" alt="Image" src="https://github.com/user-attachments/assets/71f015e1-473a-4036-a016-e31da411d86e" />

<img width="433" height="299" alt="Image" src="https://github.com/user-attachments/assets/0765a088-78e8-4508-88ef-2b8c62c9238a" />

for stunts and extras I only display the title and the description, not the short description. Do you think we should display the short description in addition ?

I'll make a pull request in the next few days, but there are quite a few warnings that I'd like to eliminate first.

You can view the code in this branch https://github.com/anvil-vtt/FateX/compare/master...Alexandre-nk-Perdereau:FateX:feature/display_aspect_stunt_extra_to_chat?expand=1

**Comment by Azaun** [comment]
Could this be adjusted so that the Aspect titles (i.e. High Concept) serves as the link that pushes the input text to chat instead of the dialogue bubble icon? The way it is setup now, the icon ends up blocking a portion of the already limited space available for the Aspect text input.

**Comment by Alexandre-nk-Perdereau** [comment]
You're right @Azaun , I will try to implement something like that.

**Comment by Azaun** [comment]
I feel I also should have added that I appreciate this feature. I was remiss in appreciating its usefulness and offering a thank you for its inclusion! It feels a worthwhile feature to me, I have seen its usefulness in play, and I appreciate your effort to include it! 
Thank you!

**Comment by Alexandre-nk-Perdereau** [comment]
@Azaun 

<img width="528" height="86" alt="Image" src="https://github.com/user-attachments/assets/05ba4e99-f99b-4989-bf89-98ba43407295" />

We have 2 option, makes the text or the label be abble to be cliked. In this example the text is "text"and the label is "Concept"

The label option is easy, but the issue is that it is often empty.

The text option is a little harde and could be counterintuitive in my opinion, but will probably never be empty.

What is your opinion?

**Comment by Azaun** [comment]
@Alexandre-nk-Perdereau 
I agree that clicking the _text_ in the example above could be counterintuitive for most users, myself included. I think I could see myself and others inadvertently passing the Aspect to chat when thinking clicking on the text of the field would start the editing process. I think clicking on the Aspects _Concept_ would be the better option. 

Would it be difficult to simply pass the label of the Aspect if the text field is empty? That way if a player clicks a blank entry, a GM would at least know which one the player is clicking and could help give them some directed prompts to help them fulfill it if they are making the character through play? Or beyond that have it post something like _Concept: Undefined_?

**Comment by Alexandre-nk-Perdereau** [comment]
@Azaun I'm not sure I understand everything you mean. Currently, with clicking the label, it works like this:

<img width="543" height="464" alt="Image" src="https://github.com/user-attachments/assets/ce1d576a-411f-4728-b96c-5ec3245162aa" />

we've a hand when hovering on the label, when we click it looks like this on the chat:

<img width="451" height="183" alt="Image" src="https://github.com/user-attachments/assets/8d82f9aa-554c-48ef-80d7-ac3b9702e4b4" />

we can also click on the empty label, it makes a hand when hovering at the very left at the place where we should see the label if it wasn't empty, let's see with the empy label but non empty text:

<img width="450" height="175" alt="Image" src="https://github.com/user-attachments/assets/dffd8e81-c5db-48da-9b3c-d192223cdc4e" />

And with an empty label and an empty text:

<img width="448" height="191" alt="Image" src="https://github.com/user-attachments/assets/9943ed5d-661c-4010-917a-3f9011052708" />

As you can see, it display aspect instead of the label when the label is empty, that is ok for me.

Is that ok for you ?

**Comment by Alexandre-nk-Perdereau** [comment]
to clarify the clickable areas it is in the red areas: 

<img width="538" height="149" alt="Image" src="https://github.com/user-attachments/assets/afa1a6ad-3de1-4bf3-a1b0-fc1f25ab1f69" />

**Comment by Azaun** [comment]
@Alexandre-nk-Perdereau 
Absolutely, this seems a fine solution. 

I misunderstood which value you meant was blank in your prior post. I thought you meant when the text was blank and not the label. I have rarely encountered an aspect without a label so I did not foresee that as being an issue as you've demonstrated it could be.

**Comment by Alexandre-nk-Perdereau** [comment]
I often label only the high concept and the problem, but not the 3 others. (the "crossing path"aspects, I often only put the name of the aspect but no label)
